addons/sale/models/sale_order_line.py

```diff
@@ -654,6 +664,34 @@ def _compute_qty_delivered(self):
         for so_line in lines_by_analytic:
             so_line.qty_delivered = mapping.get(so_line.id or so_line._origin.id, 0.0)
 
+    def _get_downpayment_state(self):
+        self.ensure_one()
+
+        if self.display_type:
+            return ""
+
+        invoice_lines = self._get_invoice_lines()
+        if all(line.parent_state == "draft" for line in invoice_lines):
+            return "draft"
+        if all(line.parent_state == "cancel" for line in invoice_lines):
+            return "cancel"
+        if all(line.parent_state == "posted" for line in invoice_lines):
+            return "posted"
+
+        return ""
```

Comment:
technical strings -> single quoted strings

Reply:
why resolved ?
'draft', 'cancel' and 'posted' are never shown to the user :thinking:

Reply:
sorry, I thought that I had changed it :/